static/src/javascripts/projects/common/modules/identity/wizard.js

```diff
@@ -31,25 +37,85 @@ const updateCounter = (wizardEl: HTMLElement): Promise<void> =>
+                10
+            ),
+            [...wizardEl.getElementsByClassName(stepClassname)],
+        ])
+        .then(
+            (
+                [
+                    offsetTop: number,
+                    currentPosition: number,
+                    steps: Array<HTMLElement>,
+                ]
+            ) =>
+                fastdom.write(() => {
+                    if (newPosition < 0 || !steps[newPosition]) {
+                        throw new Error('Invalid position');
+                    }
+                    if (currentPosition > -1 && window.scrollY > offsetTop) {
+                        scrollTo(offsetTop, 250, 'linear');
+                    }
+                    const transitionClassnames = [
+                        stepInClassname,
+                        stepInReverseClassname,
+                        stepOutClassname,
+                        stepOutReverseClassname,
+                    ];
+                    steps.forEach((stepEl: HTMLElement, i: number) => {
+                        switch (i) {
+                            case newPosition:
+                                stepEl.classList.remove(
+                                    stepHiddenClassname,
+                                    ...transitionClassnames
+                                );
+                                wizardEl.style.minHeight = `${stepEl.getBoundingClientRect()
+                                    .height}px`;
+                                if (currentPosition > -1) {
+                                    stepEl.classList.add(
+                                        currentPosition < newPosition
+                                            ? stepInClassname
+                                            : stepInReverseClassname
+                                    );
+                                }
+                                break;
+                            case currentPosition:
+                                stepEl.classList.remove(
+                                    ...transitionClassnames
+                                );
+                                stepEl.classList.add(
+                                    ...[
+                                        stepHiddenClassname,
+                                        currentPosition < newPosition
+                                            ? stepOutClassname
+                                            : stepOutReverseClassname,
+                                    ]
+                                );
+                                setTimeout(() => {
+                                    stepEl.classList.remove(
+                                        ...transitionClassnames
+                                    );
+                                }, 200);
+                                break;
+                            default:
+                                stepEl.classList.add(stepHiddenClassname);
```

Comment:
There is a lot of duplication in `.add/ .remove`. Looks like a few of these could be replaced with `.toggle()`? This would clean up the whole switch a lot I guess ...

Reply:
not really, i want to make sure the state is consistent after each composition and toggle would basically make it random

Reply:
`.toggle()` has a `force` parameter, which is really helpful in these cases:

```
toggle( String [, force] )
When only one argument is present: Toggle class value; i.e., if class exists then remove it and return false, if not, then add it and return true. When a second argument is present: If the second argument evaluates to true, add specified class value, and if it evaluates to false, remove it.
```

https://developer.mozilla.org/en-US/docs/Web/API/Element/classList